### PR TITLE
statistics: print the process of init stats

### LINKED ISSUE #53564: Showing the progress of loading initial stats
## Enhancement

Upon launch, TiDB requires the loading of extensive statistical information for query optimization and index selection, among other functions. This process can be protracted, particularly in instances of substantial data volume or intricate table structures. Users are unable to discern the progress of the loading visually and cannot ascertain its completion.

During TiDB's startup procedure, a progress bar should be displayed to indicate the progress of the statistical information loading.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #53564

Problem Summary:

### What changed and how does it work?

it sometimes takes too much time for tidb to init stats. we should print the process info to tell the users.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

if we have several tables, it will print
```
[2024/05/27 18:21:39.093 +08:00] [INFO] [bootstrap.go:734] ["complete to load the histogram"] [category=stats]
[2024/05/27 18:21:39.094 +08:00] [INFO] [bootstrap.go:743] ["complete to load the topn"] [category=stats]
[2024/05/27 18:21:39.097 +08:00] [INFO] [bootstrap.go:755] ["complete to load the bucket"] [category=stats]
[2024/05/27 18:21:39.098 +08:00] [INFO] [domain.go:2318] ["init stats info time"] [lite=false] ["take time"=13.208834ms]
```

But if TiDB has enough tables, it will print.

```
...
[2024/05/27 19:54:10.516 +08:00] [INFO] [load_stats_page.go:83] ["load bucket [1080/2391]"] [category=stats]
[2024/05/27 19:55:10.516 +08:00] [INFO] [load_stats_page.go:83] ["load bucket [1781/2391]"] [category=stats]
...
```
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
init stats 时显示进度
statistics: print the process of init stats
```

### CODE REVIEW

File: pkg/statistics/handle/initstats/load_stats_page.go [hunk 1]
```diff
@@ -57,6 +101,10 @@ func (ls *RangeWorker) loadStats() {
 		if err := ls.dealFunc(task); err != nil {
 			logutil.BgLogger().Error("load stats failed", zap.Error(err))
 		}
+		if ls.logger != nil {
+			completeTaskCnt := ls.completeTaskCnt.Add(1)
+			ls.logger.Info(fmt.Sprintf("load %s [%d/%d]", ls.taskName, completeTaskCnt, ls.taskCnt))
```
Comment (AilinKid) [inline pkg/statistics/handle/initstats/load_stats_page.go:106]: seems it's sampling output
1%    50%   90%   100% 
if 100% is missed，will the log msg looks wired?

**Review by AilinKid — APPROVED** [review]
Rest LGTM

File: pkg/statistics/handle/initstats/load_stats_page.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/statistics/handle/initstats/load_stats_page.go:106]: If it is complete, It must print ```complete```.

Comment (hawkingrei) [reply pkg/statistics/handle/initstats/load_stats_page.go:106]: load xxxx [completed task count / total task count]

**Comment by hawkingrei** [comment]
@winoros @hi-rustin PTAL

File: pkg/statistics/handle/bootstrap.go [hunk 2]
```diff
@@ -697,10 +698,12 @@ func (h *Handle) InitStatsLite(is infoschema.InfoSchema) (err error) {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	statslogutil.StatsLogger().Info("complete to load the meta")
 	err = h.initStatsHistogramsLite(is, cache)
 	if err != nil {
 		return errors.Trace(err)
 	}
+	statslogutil.StatsLogger().Info("complete to load the histogram")
```
Comment (elsa0520) [inline pkg/statistics/handle/bootstrap.go:706]:
```suggestion
	statslogutil.StatsLogger().Info("complete to load the lite histogram")
```

**Review by winoros — APPROVED** [review]

File: pkg/statistics/handle/bootstrap.go (hunk 2 above)
Comment (hawkingrei) [reply pkg/statistics/handle/bootstrap.go]: updated.

**Comment by hawkingrei** [comment]
@elsa0520 Please confirm one last time.

**Review by elsa0520 — APPROVED** [review]
LGTM

**Pull request merged**